codalab/lib/upload_manager.py

```diff
@@ -166,20 +166,18 @@ def update_metadata_and_save(self, bundle, new_bundle):
 
         data_hash = '0x%s' % (path_util.hash_directory(bundle_path, dirs_and_files))
         data_size = path_util.get_size(bundle_path, dirs_and_files)
+        if enforce_disk_quota:
+            disk_left = self._bundle_model.get_user_disk_quota_left(bundle.owner_id)
+            if data_size > disk_left:
+                raise UsageError("Can't save bundle, bundle size %s greater than user's disk quota left: %s" % (data_size, disk_left))
```

Comment:
does a bundle that gets expanded beyond the disk quota actually fall into a fail state after reaching this point?

Reply:
We have that logic builtin for run bundles. Run bundles fail as soon as their data size hits `request_disk`. This change prevents upload bundles from overcoming the limit.

I'd say this is not the best way of doing it for upload bundles as this function only gets called after the upload is done. However trying to enforce the quota at every possible upload code path would be very messy (especially in places where we just call command line programs like git clone). So this was the second best way I could think of as at least it cleans up the disk afterwards.